Wrong HAL family in find_package

**Describe the bug**
When building from a clean project and fetching multiple HAL families CMake Error happens. For example, when trying to build fetch-cmsis-hal example:

```
CMake Error at /<path...>/stm32-cmake/cmake/FindHAL.cmake:79 (message):
  could not find HAL for family F4
Call Stack (most recent call first):
  CMakeLists.txt:10 (find_package)
```
If i add following code:
```
# Step 2 : Generating all the valid drivers from requested families
...
# added portion
    message("${FAMILY}")
    message("${FAMILY_L}")
# added portion over

    find_path(HAL_${FAMILY}_PATH
```
The output from it is:
```
F4
l0
```

After i added FAMILY_L assignment the problem was solved
```
    if(CMAKE_MATCH_1) #Matches the family part of the provided STM32<FAMILY>[..] component
        set(FAMILY ${CMAKE_MATCH_1})
        string(TOLOWER ${FAMILY} FAMILY_L) # my addition
    endif()
```

**Sources to reproduce**
The default fetch-cmsis-hal example is enough to reproduce

**Steps to reproduce**
1. Go to '/.../stm32-cmake/examples/fetch-cmsis-hal'
2. Type 'cmake .'
3. See error

**Expected behavior**
Successful build.

**Environment (please complete the following information):**
 - OS: Ubuntu-22.04 in WSL2
 - Compiler: arm-none-eabi-gcc 10.3.1
 - stm32-cmake: 6d07cb82
 - cmake: 3.22.1
 - HAL/cube/CMSIS: auto fetch

--- a/cmake/FindHAL.cmake
+++ b/cmake/FindHAL.cmake
@@ -62,6 +62,7 @@ foreach(family_comp ${HAL_FIND_COMPONENTS_FAMILIES})
     string(REGEX MATCH "^STM32([FGHLMUW]P?[0-9BL])([0-9A-Z][0-9M][A-Z][0-9A-Z])?_?(M0PLUS|M4|M7)?.*$" family_comp ${family_comp})
     if(CMAKE_MATCH_1) #Matches the family part of the provided STM32<FAMILY>[..] component
         set(FAMILY ${CMAKE_MATCH_1})
+        string(TOLOWER ${FAMILY} FAMILY_L)
     endif()
     find_path(HAL_${FAMILY}_PATH
         NAMES Inc/stm32${FAMILY_L}xx_hal.h

